src/nix-instantiate/nix-instantiate.cc

```diff
@@ -54,6 +54,9 @@ void processExpr(EvalState & state, const Strings & attrPaths,
             else if (output == okJSON) {
                 printValueAsJSON(state, strict, vRes, v.determinePos(noPos), std::cout, context);
                 std::cout << std::endl;
+            } else if (output == okRaw) {
+                std::cout << *state.coerceToString(noPos, vRes, context, "while generating the nix-instantiate output");
+                std::cout << std::endl;
```

Comment:
`nix eval --raw` is even more raw, by not adding a trailing newline.

> ```c++
>             writeFull(STDOUT_FILENO, *state->coerceToString(noPos, *v, context, "while generating the eval command output"));
> ```

I think in both commands, we should make the extra newline dependent on whether `stdout` is a tty, probably as a general rule to be implemented in the proposed (elsewhere) function that factors out the `stopProgressBar()` + `writeFull(STDOUT...` combination.

Reply:
Also I see no reason why we couldn't stop the progress bar in the old CLI.
If we don't have one, stopping it should just be a no-op.